base/error.jl

```diff
@@ -52,28 +52,31 @@ end
 
 
 """
-    retry(f, [condition]; n=3; max_delay=10) -> Function
+    retry(f, [on_error]; n=1; max_delay=0) -> Function
 
 Returns a lambda that retries function `f` up to `n` times in the
-event of an exception. If `condition` is a `Type` then retry only
-for exceptions of that type. If `condition` is a function
-`cond(::Exception) -> Bool` then retry only if it is true.
+event of an exception. If `on_error` is a `Type` then retry only
+for exceptions of that type. If `on_error` is a function
+`test_error(::Exception) -> Bool` then retry only if it is true.
+If unspecified, retry for all errors.
```

Comment:
> If unspecified, retry for all errors.

This seems redundant because we already say _"... retries ... in the
 -event of an exception."_ and the `[` and `]` indicate that `condition` is optional.